igor-web/src/main/groovy/com/netflix/spinnaker/igor/wercker/WerckerService.groovy

```diff
@@ -0,0 +1,349 @@
+        Pipeline pipeline = pipelines.find {p -> pipelineName.equals(p.name)}
+        if (pipeline) {
+            log.info("Triggering run for pipeline {} with id {}",
+                    kv("pipelineName", pipelineName), kv("pipelineId", pipeline.id))
+            try {
+                Map<String, Object> runInfo = werckerClient.triggerBuild(
+                        authHeaderValue, new RunPayload(pipeline.id, 'Triggered from Spinnaker'))
+                //TODO desagar the triggerBuild call above itself returns a Run, but the createdAt date
+                //is not in ISO8601 format, and parsing fails. The following is a temporary
+                //workaround - the getRunById call below gets the same Run object but Wercker
+                //returns the date in the ISO8601 format for this case.
+                Run run = werckerClient.getRunById(authHeaderValue, runInfo.get('id'))
+
+                //Create an entry in the WerckerCache for this new run. This will also generate
+                //an integer build number for the run
+                Map<String, Integer> runIdBuildNumbers = cache.updateBuildNumbers(
+                        master, appAndPipelineName, Collections.singletonList(run))
+
+                log.info("Triggered run {} at URL {} with build number {}",
+                        kv("runId", run.id), kv("url", run.url),
+                        kv("buildNumber", runIdBuildNumbers.get(run.id)))
+
+                //return the integer build number for this run id
+                return runIdBuildNumbers.get(run.id)
+            } catch (RetrofitError e) {
+                def body = e.getResponse().getBody()
+                String wkrMsg
+                if (body instanceof TypedByteArray) {
+                    wkrMsg = new String(((retrofit.mime.TypedByteArray) body).getBytes())
+                } else {
+                    wkrMsg = body.in().text
+                }
+                log.error("Failed to trigger build for pipeline {}. {}", kv("pipelineName", pipelineName), kv("errMsg", wkrMsg))
+                throw new BuildController.BuildJobError(
+                "Failed to trigger build for pipeline ${pipelineName}! Error from Wercker is: ${wkrMsg}")
+            }
+        } else {
+            throw new BuildController.InvalidJobParameterException(
+            "Could not retrieve pipeline ${pipelineName} for application ${appName} from Wercker!")
+        }
+    }
+
+    /**
+     * Returns List of all Wercker jobs in the format of type/org/app/pipeline 
+     */
+    public List<String> getJobs() {
+        List<String> jobs = []
+        long start = System.currentTimeMillis();
+        List<Application> apps = getApplications();
+        start = System.currentTimeMillis()
+        apps.each { app ->
+            try {
+                List<Pipeline> pipelines = app.pipelines? app.pipelines : []; //getPipelines(app.owner.name, app.name);
```

Comment:
In Groovy this can be just `app.pipelines ?: []` (the "Elvis operator")

Reply:
done